### PR TITLE
Document all collected data from stats reporter

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description

This PR improve the transparency of the usage stats reporter detailing the
information collected by the agent when the stats reporter it's enabled. 

#### PR Checklist

- [ ] CHANGELOG updated
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: docs/sources/configuration/flags.md [hunk 1]
```diff
@@ -43,6 +43,14 @@ These statistics are sent to `stats.grafana.org`.
 Statistics help us better understand how Grafana Agent is used.
 This helps us prioritize features and documentation.
 
+Besides the enabled feature flags we are also collecting the following information:
+* A randomly generated and an anonymous unique ID (UUID).
+* Timestamp of when the UID was first generated.
+* Timestamp of when the report was created (by default every 4h).
+* Version of running Grafana Agent.
+* Operating system Grafana Agent is running on.
+* System architecture Grafana Agent is running on.
```
Comment (rfratto) [inline docs/sources/configuration/flags.md:52]: IMO I'd change the first paragraph to not mention the feature flags:

```
By default, Grafana Agent sends anonymous, but uniquely-identifiable usage information for your running Grafana Agent instance to Grafana Labs.
These statistics are sent to `stats.grafana.org`.
``` 

And then change this to be a fully exhaustive list: 

```
The usage information includes the following details:
* A randomly generated and an anonymous unique ID (UUID).
* Timestamp of when the UID was first generated.
* Timestamp of when the report was created (by default every 4h).
* Version of running Grafana Agent.
* Operating system Grafana Agent is running on.
* System architecture Grafana Agent is running on.
* List of enabled feature flags. 
```

I would also add a disclaimer that this list may change over time, and if they do, we will describe changes in the changelog.

File: docs/sources/configuration/flags.md [hunk 2]
```diff
@@ -34,15 +34,26 @@ Valid feature names are:
 * `dynamic-config`: Enable support for [dynamic configuration]({{< relref "./dynamic-config" >}})
 * `extra-scrape-metrics`: When enabled, additional time series  are exposed for each metrics instance scrape. See [Extra scrape metrics](https://prometheus.io/docs/prometheus/latest/feature_flags/#extra-scrape-metrics).
 
-### Report use of feature flags
+## Report information usage
 
-By default, Grafana Agent sends anonymous, but uniquely-identifiable information
-of the enabled feature flags from your running Grafana Agent instance to Grafana Labs.
+By default, Grafana Agent sends anonymous, but uniquely-identifiable usage information
+for your running Grafana Agent instance to Grafana Labs.
```
Comment (captncraig) [inline docs/sources/configuration/flags.md:40]: "from your running" not "for"

File: docs/sources/configuration/flags.md [hunk 3]
```diff
@@ -34,15 +34,26 @@ Valid feature names are:
...
 These statistics are sent to `stats.grafana.org`.
 
 Statistics help us better understand how Grafana Agent is used.
 This helps us prioritize features and documentation.
 
+The usage information includes the following details:
+* A randomly generated and an anonymous unique ID (UUID).
+* Timestamp of when the UID was first generated.
+* Timestamp of when the report was created (by default every 4h).
```
Comment (karengermond) [inline docs/sources/configuration/flags.md:49]:
```suggestion
* Timestamp of when the report was created (by default, every 4h).
```

File: docs/sources/configuration/flags.md [hunk 4]
```diff
@@ -34,15 +34,26 @@ Valid feature names are:
...
+* Version of running Grafana Agent.
+* Operating system Grafana Agent is running on.
+* System architecture Grafana Agent is running on.
+* List of enabled feature flags.
+
+This list may change over time. All newly reported data will be also documented in the CHANGELOG.
```
Comment (karengermond) [inline docs/sources/configuration/flags.md:55]:
```suggestion
This list may change over time. All newly reported data will also be documented in the CHANGELOG.
```

**Review by karengermond — APPROVED** [review]
LGTM!

**Pull request merged**